kernel: bump 6.6 to 6.6.25

graysky2 opened the pull request:
kernel: bump 6.6 to 6.6.25
    
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.25
    
All patches automatically rebased.
    
Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000

graysky2 committed:
kernel: bump 6.6 to 6.6.24

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.24

Removed upstreamed:
	generic/backport-6.6/838-v6.9-leds-trigger-netdev-Fix-kernel-panic-on-interface-re.patch[1]
	generic/backport-6.6/981-mtd-spinand-Add-support-for-5-byte-IDs.patch[2]
	mediatek/patches-6.6/831-thermal-drivers-mediatek-Fix-control-buffer-enablement-on-MT7896.patch[3]

Removed:
	ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch[4]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.24&id=10f2af1af8ab8a7064f193446abd5579d3def7e3
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.24&id=9c74507e6c4382d12a5e418742b81fd441f03313
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.24&id=fe750e2744428be6aec7395b13df3b34381ce6b6
4. Acknowledgment to @DragonBlurp for pointing out the redundancy of this patch. https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
@Ansuel Please check my manual rebase of your patch.  I changed:
```
config.add_legacy_fixed_of_cells = of_device_is_compatible(node, "nvmem-cells");
```
to
```
config.add_legacy_fixed_of_cells = !of_node_name_eq(mtd->dev.of_node, "nand");
```

zxlhhyccc commented:
@graysky2 Is kernel 6.1 no longer updated?

DragonBluep commented on target/linux/ipq40xx/patches-6.6/110-mtd-limit-OTP-nvmem-to-non-nand-devices.patch:
```diff
@@ -47,12 +47,12 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
 +++ b/drivers/mtd/mtdcore.c
-@@ -933,7 +933,7 @@ static struct nvmem_device *mtd_otp_nvme
- 	config.name = compatible;
- 	config.id = NVMEM_DEVID_AUTO;
```
I think this patch can be removed as it's the same fix as https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807.

graysky2 replied:
Thanks, fixed.

namiltd replied:
My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363
```
-	config.add_legacy_fixed_of_cells = !!np;
+	config.add_legacy_fixed_of_cells = !!np && !of_node_name_eq(mtd->dev.of_node, "nand");

namiltd replied:
My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363
```
-	config.add_legacy_fixed_of_cells = !!np;
+	config.add_legacy_fixed_of_cells = !!np && !of_node_name_eq(mtd->dev.of_node, "nand");

namiltd commented:
PR https://github.com/openwrt/openwrt/pull/14868 is merged. Please check if your patch requires an update

namiltd reviewed (commented):
My proposal combining the fix https://github.com/openwrt/openwrt/commit/dd78a59cd7b029560b33cb3ac0e1aa8b747bd807 and https://github.com/openwrt/openwrt/pull/14940/commits/8ec76705d42766156c4185f59998c7144f815363

graysky2 committed:
kernel: bump 6.6 to 6.6.25

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.25

All patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000

Signed-off-by: John Audia <therealgraysky@proton.me>

namiltd commented:
PR https://github.com/openwrt/openwrt/pull/1477 merged, Please check if your patch requires an update.
